subiquity/server/controllers/filesystem.py

```diff
@@ -887,8 +886,7 @@ async def _probe_once(self, *, context, restricted):
                 probe_types |= {'os'}
             fname = 'probe-data.json'
             key = "ProbeData"
-        storage = await run_in_thread(
-            self.app.prober.get_storage, probe_types)
+        storage = await self.app.prober.get_storage(probe_types)
```

Comment:
Well, I made `get_storage` responsible from spawning the thread so probert should still execute in a separate thread ; unless I'm overlooking something.

The only difference AFAICT is that the other instructions in `get_storage` will now execute in the main thread. I think it is safer since currently, we have assignments to `self` from multiple threads without locking.